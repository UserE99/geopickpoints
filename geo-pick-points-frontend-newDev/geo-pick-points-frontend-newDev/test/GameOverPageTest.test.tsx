import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import GameOverPage from "../src/Pages/GameOverPage";
import { Provider } from "react-redux";
import { MemoryRouter, useNavigate, useLocation } from "react-router-dom";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

describe("GameOverPage Component", () => {
  let store;
  let mockNavigate;

  beforeEach(() => {
    store = mockStore({}); 
    mockNavigate = jest.fn();
    (useNavigate as jest.Mock).mockReturnValue(mockNavigate);

    (useLocation as jest.Mock).mockReturnValue({
      state: {
        dataGameInstance: {
          payload: { startTime: new Date(Date.now() - 60000).toISOString() }, 
        },
        scores: new Map([
          ["Team A", 100],
          ["Team B", 80],
        ]),
      },
    });
  });

  it("renders GameOverPage correctly and matches snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays game time correctly if available", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/Time:/i)).toBeInTheDocument();
    expect(screen.getByText(/1 minutes and \d{1,2} seconds/i)).toBeInTheDocument();
  });

  it("displays 'Not available' ", () => {
    (useLocation as jest.Mock).mockReturnValue({
      state: {
        dataGameInstance: { payload: {} }, 
        scores: new Map(),
      },
    });

    render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/Time:/i)).toBeInTheDocument();
    expect(screen.getByText(/Not available/i)).toBeInTheDocument();
  });

  it("displays team scores correctly", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/Scores/i)).toBeInTheDocument();
    expect(screen.getByText(/Team A - Score 100/i)).toBeInTheDocument();
    expect(screen.getByText(/Team B - Score 80/i)).toBeInTheDocument();
  });

  it("shows 'No scores available'", () => {
    (useLocation as jest.Mock).mockReturnValue({
      state: {
        dataGameInstance: { payload: {} },
        scores: new Map(), 
      },
    });

    render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/No scores available/i)).toBeInTheDocument();
  });

  it("navigates back to start ", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <GameOverPage />
        </MemoryRouter>
      </Provider>
    );

    const backButton = screen.getByRole("button", { name: /Back to Start/i });
    fireEvent.click(backButton);

    expect(mockNavigate).toHaveBeenCalledWith("/");
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });
});

