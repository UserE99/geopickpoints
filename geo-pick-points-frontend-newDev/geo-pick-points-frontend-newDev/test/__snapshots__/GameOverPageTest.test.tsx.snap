// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverPage Component renders GameOverPage correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding-top: 20vh; text-align: center;"
  >
    <h1>
      Game Over
    </h1>
    <p>
      <strong>
        Time:
      </strong>
       1 minutes and 0 seconds
    </p>
    <div>
      <h2>
        Scores
      </h2>
      <ul
        style="list-style-type: none; padding: 0px;"
      >
        <li
          style="margin: 10px 0px;"
        >
          <strong>
            Team A
          </strong>
           - Score 100
        </li>
        <li
          style="margin: 10px 0px;"
        >
          <strong>
            Team B
          </strong>
           - Score 80
        </li>
      </ul>
    </div>
    <button
      style="margin-top: 20px; padding: 10px 20px; font-size: 16px; cursor: pointer; border: solid; border-radius: 5px;"
    >
      Back to Start
    </button>
  </div>
</DocumentFragment>
`;
